src/uthash.h

```diff
@@ -323,13 +375,24 @@ do {
     (add)->hh.prev = NULL;                                                       \
     (head) = (add);                                                              \
     HASH_MAKE_TABLE(hh, head);                                                   \
+    if (HASH_NOMEM_OK && !(head)->hh.tbl) {                                      \
+      (head) = 0;                                                                \
```

Comment:
Lines 376–379 could almost certainly be refactored to `HASH_MAKE_TABLE(hh, add)` and then assign `(head) = (add)` only if the `HASH_MAKE_TABLE` succeeded.